src/libraries/System.Security.Cryptography.Algorithms/ref/System.Security.Cryptography.Algorithms.cs

```diff
@@ -186,14 +186,14 @@ public partial struct ECCurve
         private int _dummyPrimitive;
         public byte[] A;
         public byte[] B;
```

Comment:
What is the rule we're using with structs that are used as input?  For ECPoint I answered an outstanding "should these be marked nullable" with yes because there are times that callers build the empty struct and assign the fields, then pass it as input.  But ECCurve, RSAParameters, and DSAParameters marked the required parameters as non-null and the optional ones as nullable.

If "you must set this" (and "we will never return this as null") means "non-null", then ECPoint's X and Y are both non-null.  If "we expect you to create this as the empty struct then populate fields" is means "null", then all of the reference types deeply on RSAParameters, ECParameters and DSAParameters should be nullable.

Reply:
I think that is the case mark it nullable and add [DisallowNull] attribute, but still we will need lots of bang for making them nullable, @stephentoub thoughts?

Reply:
I think it's weird for a field to be nullable with [DisallowNull]; that means you can get null out but can't set it to null, but as it's a field, there's no processing logic, so having such a configuration is... strange.  If the expectation with ECPoint is that you may create the default struct and then populate some fields, that also means any consumer would need to check the fields before using them, so just having them be nullable would seem to make sense. No?

Reply:
Well nullable to express it can be null until you set, and `[DisallowNull]` to force that it cannot be set with null. But yeah doesn't make much sense though

Reply:
So it sounds like all of the reference types on RSAParameters, DSAParameters, ECParameters, ECCurve, and ECPoint (done) should be nullable, since they all have expectations of being caller-populated input types?

e.g.

```C#
RSAParameters rsaParameters = new RSAParameters
{
    Modulus = ...,
    Exponent = ...,
};

rsa.ImportParameters(rsaParameters);

// Using an accelerated curve value here, vs an explicitly defined curve from many parameters;
// but the hard way is also possible.
ECParameters ecParameters = new ECParameters
{
    Curve = ECCurve.NamedCurves.nistP256,
    Q = new ECPoint
    {
        X = ...,
        Y = ...,
    },
};

ecdsa.ImportParameters(ecParameters);
```

Reply:
Yep, updated